sympy/functions/special/gamma_functions.py

```diff
@@ -198,6 +198,25 @@ def _sage_(self):
         import sage.all as sage
         return sage.gamma(self.args[0]._sage_())
 
+    def _eval_as_leading_term(self, x):
+        from sympy import Order
+        arg = self.args[0]
+        arg_1 = arg.as_leading_term(x)
+        if Order(x, x).contains(arg_1):
+            return None
```

Comment:
The gamma function has a simple pole with residue 1 at the origin. This could probably be `return 1/arg_1`.

Reply:
Changes done. Also some tests added.